# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsGlobalWindowInner.cpp
# Commit: 08eb9cf235bb
# Full Hash: 08eb9cf235bb5334ef01075d6148e1ea235adf56
# Author: Benjamin VanderSloot <bvandersloot@mozilla.com>
# Date: 2025-06-05 04:07:37
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D251617
# ==============================================================================

diff -r ff3135366145 -r 08eb9cf235bb dom/base/nsGlobalWindowInner.cpp
--- a/dom/base/nsGlobalWindowInner.cpp	Wed Jun 04 19:25:54 2025 +0000
+++ b/dom/base/nsGlobalWindowInner.cpp	Wed Jun 04 19:25:54 2025 +0000
@@ -183,6 +183,7 @@
 #include "mozilla/dom/VRDisplayEventBinding.h"
 #include "mozilla/dom/VREventObserver.h"
 #include "mozilla/dom/VisualViewport.h"
+#include "mozilla/dom/WebIdentityHandler.h"
 #include "mozilla/dom/WebIDLGlobalNameHash.h"
 #include "mozilla/dom/WindowBinding.h"
 #include "mozilla/dom/WindowContext.h"
@@ -7790,6 +7791,18 @@
   return wc->GetUsingStorageAccess();
 }
 
+WebIdentityHandler* nsPIDOMWindowInner::GetOrCreateWebIdentityHandler() {
+  if (mWebIdentityHandler) {
+    return mWebIdentityHandler;
+  }
+  mWebIdentityHandler = new WebIdentityHandler(this);
+  bool success = mWebIdentityHandler->MaybeCreateActor();
+  if (!success) {
+    mWebIdentityHandler = nullptr;
+  }
+  return mWebIdentityHandler;
+}
+
 CloseWatcherManager* nsPIDOMWindowInner::EnsureCloseWatcherManager() {
   if (!mCloseWatcherManager) {
     mCloseWatcherManager = new CloseWatcherManager();